java/maven/src/org/netbeans/modules/maven/runjar/MavenExecuteUtils.java

```diff
@@ -0,0 +1,859 @@
+        PropertySplitter ps = new PropertySplitter(line);
+        ps.setSeparator(' '); //NOI18N
+        String s = ps.nextPair();
+        while (s != null) {
+            if (s.startsWith("-") || s.contains("%classpath")) { //NOI18N
+                s = ps.nextPair();
+                continue;
+            } else if (s.equals(PACKAGE_CLASS_NAME_TOKEN) || s.equals(RUN_MAIN_CLASS_TOKEN) || s.matches("[\\w]+[\\.]{0,1}[\\w\\.]*")) { //NOI18N
+                return s;
+            } else {
+                Logger.getLogger(RunJarPanel.class.getName()).fine("failed splitting main class from=" + line); //NOI18N
+            }
+            s = ps.nextPair();
+        }
+        return ""; //NOI18N
+    }
+    
+    @NonNull
+    public static String splitParams(String line) {
+        int argsIndex = line.indexOf(RunJarStartupArgs.USER_PROGRAM_ARGS_MARKER);
+        if (argsIndex > -1) {
+            return line.substring(argsIndex + RunJarStartupArgs.USER_PROGRAM_ARGS_MARKER.length()).trim();
+        }
+        String main = splitMainClass(line);
+        if (main.isEmpty()) {
+            return "";
+        }
+        int i = line.indexOf(main);
+        if (i > -1) {
+            return line.substring(i + main.length()).trim();
+        }
+        return ""; //NOI18N
+    }
+    /**
+     * Splits the line into sequence of arguments, respects quoting.
+     * @param line the line as a string
+     * @return arguments in an iterable
+     */
+    public static Iterable<String> propertySplitter(String line) {
+        return propertySplitter(line, true);
+    }
+    
+    public static Iterable<String> propertySplitter(String line, boolean outputQuotes) {
+        class SplitIt implements Iterator<String> {
+            private final PropertySplitter spl = new PropertySplitter(line);
+            private String nextPair;
+
+            public SplitIt() {
+                spl.setSeparator(' ');
+                spl.setOutputQuotes(outputQuotes);
+            }
+            
+            @Override
+            public boolean hasNext() {
+                if (nextPair == null) {
+                    nextPair = spl.nextPair();
+                }
+                return nextPair != null;
+            }
+
+            @Override
+            public String next() {
+                String s;
+                if (nextPair == null) {
+                    nextPair = spl.nextPair();
+                }
+                s = nextPair;
+                nextPair = null;
+                if (s != null) {
+                    return s;
+                } else {
+                    throw new NoSuchElementException();
+                }
+            }
+        }
+        return new Iterable<String>() {
+            @Override
+            public Iterator<String> iterator() {
+                return new SplitIt();
+            }
+        };
+    }
+    
+    /**
+     *
+     * @author mkleint
+     */
+    static class PropertySplitter {
```

Comment:
java/maven/src/org/netbeans/modules/maven/customizer/PropertySplitter.java is not deleted.

Reply:
Property splitter unified in 50ee5d6